dashboard/src/reducers/functions.ts

```diff
@@ -31,7 +31,11 @@ const functionsReducer = (
       return { ...state, isFetching: true };
     case getType(actions.functions.selectFunction):
       const { f } = action;
-      return { ...state, isFetching: false, selected: { ...state.selected, function: f } };
+      return {
+        ...state,
+        isFetching: false,
+        selected: { ...state.selected, function: f, podName: undefined },
```

Comment:
without this, the log view will try to initially load logs from old state. e.g. after visiting functionA, you go back to the list and then click on functionB, because the state wasn't cleared before, podName would still point to functionA.